rspace/src/main/scala/coop/rchain/rspace/history/KeyPath.scala

```diff
@@ -0,0 +1,42 @@
+package coop.rchain.rspace.history
+
+import scodec.bits.ByteVector
+
+sealed trait KeyPath {
+  def value: ByteVector
+  def size: Long                  = value.size
+  def nonEmpty: Boolean           = value.nonEmpty
+  def head: Byte                  = value.head
+  def tailToValue: ByteVector     = value.tail
+  def tail: KeyPath               = KeyPath(tailToValue)
+  def ++(other: KeyPath): KeyPath = KeyPath(value ++ other.value)
+  def ==(other: KeyPath): Boolean = value == other.value
```

Comment:
Equality should be defined following Java guidelines which will then be part of built in `==` operator.

```scala
trait MyTrait[A] {
  val v: A

  override def equals(obj: Any): Boolean =
    obj match {
      case c: MyTrait[A] => c.v == v
      case _             => false
    }

  override def hashCode(): Int = v.hashCode()
}

val a1 = new MyTrait[Int] {
  val v = 42
}

val a2 = new MyTrait[Int] {
  val v = 42
}

val (a1Eqa2, a2Eqa1) = (a1 == a2, a2 == a1)
val (h1, h2) = (a1.hashCode(), a2.hashCode())
```